src/react/ssr/__tests__/prerenderStatic.test.tsx

```diff
@@ -0,0 +1,777 @@
+/* eslint-disable testing-library/render-result-naming-convention */
+import "../../../testing/internal/messageChannelPolyfill.js";
+
+import "./polyfillReactDomTypes.d.ts";
+
+import { expectTypeOf } from "expect-type";
+import { JSDOM } from "jsdom";
+import jsesc from "jsesc";
+import * as React from "react";
+import { renderToStaticMarkup, renderToString } from "react-dom/server";
+import { prerender } from "react-dom/static";
+import { prerenderToNodeStream } from "react-dom/static.node";
+
+import {
+  ApolloClient,
+  ApolloLink,
+  gql,
+  InMemoryCache,
+  TypedDocumentNode,
+} from "@apollo/client/core";
+import {
+  ApolloProvider,
+  getApolloContext,
+  useQuery,
+  useSuspenseQuery,
+} from "@apollo/client/react";
+import { prerenderStatic } from "@apollo/client/react/ssr";
+import {
+  MockedResponse,
+  MockLink,
+  MockSubscriptionLink,
+  wait,
+} from "@apollo/client/testing";
+import { InvariantError } from "@apollo/client/utilities/invariant";
+
+import { resetApolloContext } from "../../../testing/internal/resetApolloContext.js";
+
+beforeEach(() => {
+  // We are running tests with multiple different renderers, and that can result in a warning like
+  // > Detected multiple renderers concurrently rendering the same context provider. This is currently unsupported.
+  // This avoids that.
+  resetApolloContext();
+});
+// @ts-ignore
+global.setImmediate ||= (fn) => setTimeout(fn, 0);
+
+function testSetup() {
+  const query1: TypedDocumentNode<{ hello: string }> = gql`
+    query {
+      hello
+    }
+  `;
+  const query2: TypedDocumentNode<{ whoami: { name: string } }> = gql`
+    query {
+      whoami {
+        name
+      }
+    }
+  `;
+  const query3: TypedDocumentNode<{ currentTime: string }> = gql`
+    query {
+      currentTime
+    }
+  `;
+
+  function Outlet() {
+    const { data } = useSuspenseQuery(query1);
+    return (
+      <div>
+        <p>Hello {data.hello}!</p>
+        <Parent />
+      </div>
+    );
+  }
+
+  function Parent() {
+    const { data } = useSuspenseQuery(query2);
+
+    return (
+      <>
+        <p>My name is {data?.whoami.name}!</p>
```

Comment:
```suggestion
        <p>My name is {data.whoami.name}!</p>
```

Shouldn't this be a non-null value?